include/neml2/tensors/Scalar.h

```diff
@@ -34,27 +34,21 @@ class Scalar : public FixedDimTensor<1, 1>
   /// Forward all the constructors
   using FixedDimTensor<1, 1>::FixedDimTensor;
 
+  Scalar(Real) = delete;
```

Comment:
This is also important... I mean, intuitively we ought to be careful here, because our "Scalar" is not the scalar in torch's sense where they have compile time cuda kernels implemented (and so they don't have to move the scalar to different devices at run time).

Reply:
I guess I see the logic in having just the `Scalar(torch::tensor)` type constructor.